crates/rpc/rpc/src/debug.rs

```diff
@@ -699,6 +717,72 @@ where
         Ok(exec_witness)
     }
 
+    /// Generates a flat execution witness, using the given recovered block.
+    pub async fn debug_flat_execution_witness_for_block(
+        &self,
+        block: Arc<RecoveredBlock<ProviderBlock<Eth::Provider>>>,
+    ) -> Result<FlatExecutionWitness, Eth::Error> {
+        let this = self.clone();
+        let block_number = block.header().number();
+
+        let (pre_state, lowest_block_number) = self
```

Comment:
This should look quite similar/analogous to the `debug_execution_witness`.
The main difference is that we do `FlatWitnessRecord::record_executed_state` rather than `WitnessREcord::record_executed_state`.